### PR TITLE
[WIP] Conda recipes improvements

### PR DESCRIPTION
**Pull request by marcinz**
Ongoing work on conda recipes (update PR description before merging).

### CODE REVIEW

File: conda/conda-build/conda_build_config.yaml [hunk 1]
```diff
@@ -1,8 +1,6 @@
-# Conda only accepts strings as variant configs
-# so use an empty string "" to denote a falsey value
 gpu_enabled:
   - true
-  - 
+  - false
```
Comment (bryevdv) [inline conda/conda-build/conda_build_config.yaml:3]: cc @m3vaz is this conda issue no longer an issue?

Comment (marcinz) [reply conda/conda-build/conda_build_config.yaml:3]: It's still an issue in the sense that you can't have true boolean variables. After seeing the problems this caused for you, I thought we could just use an explicit string for false.

Comment (m3vaz) [reply conda/conda-build/conda_build_config.yaml:3]: @bryevdv nope. Marcin insisted on the magic words. I've tried to mitigate the possible damage by defining a local variable in the jinja template. 
```
{% if gpu_enabled == "true" %}
  {% set gpu_enabled_bool = true %}
{% elif gpu_enabled == "false" %}
  {% set gpu_enabled_bool = false %}
{% else %}
  {# We need to have a default value for the initial pass over the recipe #}
  {% set gpu_enabled_bool = false %}
{% endif %}
```

**Review by m3vaz — APPROVED** [review]

**Comment by marcinz** [comment]
This PR got accidentally merged with a `[WIP]` tag.

**Pull request merged**